python/ray/autoscaler/_private/spark/spark_job_server.py

```diff
@@ -181,6 +201,7 @@ def shutdown(self) -> None:
         super().shutdown()
         for spark_job_group_id in self.task_status_dict:
             self.spark.sparkContext.cancelJobGroup(spark_job_group_id)
+        time.sleep(1)  # wait for all spark job cancellation
```

Comment:
this 1s sleep is a bit arbitrary, can we remove it?

Reply:
On Databricks platform, when detaching a notebook, it triggers the "shutdown" invocation, but if it doesn't wait for about 1s, the job cancellation might fail.

Reply:
Do you know exactly why the job cancellation might fail without sleep?

Reply:
The spark job cancellation will do things asyncly in a background thread, On Databricks platform, when detaching a notebook,  it triggers SIGTERM and then sigterm handler triggers Ray cluster shutdown, without sleep, after the SIGTERM handler execution the process is killed and then cancelling spark job background threads are killed.

Reply:
Instead of sleep, can we use some signal to wait until the background thread finishes and job  is cancelled which is more deterministic. A random 1s is too arbitrary.

Reply:
We can register `SparkListenerJobEnd` event listener to do this. But it needs to write Scala code which is infeasible to integrate with Ray code.
so for now I think we have to use `sleep`

Reply:
I think it is not a big deal , in my test it works well.

Reply:
Can you convert your reply to comment saying why we have to sleep

Reply:
Sure.